firmware/ventilator-controller-stm32/Core/Test/Pufferfish/Protocols/messages.txt

```diff
@@ -0,0 +1,181 @@
+      THEN("After the write method is called, the output buffer remains unchanged")
+
+    WHEN("The message payload tag is such that it corresponds to Unrecognized message descriptor in descriptor array")
+      THEN("The write method reports invalid type status")
+      THEN("After the write method is called, the output buffer remains unchanged")
+
+    WHEN("The output buffer is too small to hold the generated body")
+      THEN("The write method reports invalid length status")
+      THEN("After the write method is called, the output buffer remains unchanged")
+
+    WHEN("The size of the output buffer is zero")
+      THEN("The write method reports invalid length status")
+
+    // Intent: To show that the write method does not change the internal "type" member of the message class
+    WHEN("The internal type member is set to an arbitrary value and then write method is called")
+    // setup:
+    // type is set to 4
+    // A sensor measurements payload message is set, which has value of type as 2
+      THEN("The write method reports ok status")
+      THEN("After the write method is called, type member remains unchanged)
+
+
+  GIVEN("A Message object constructed with StateSegment Taggedunion, default message types and a max size of 126 bytes")
+    WHEN("The payload is a sensor measurements message and write method is called")
+      THEN("The write method reports ok status")
+      THEN("The body's header correctly stores the value of message type obtained from the message payload tag")
+      THEN("The output buffer is as expected")
+
+  GIVEN("A Message object constructed with StateSegment Taggedunion, default message types and a max size of 508 bytes")
+    WHEN("The sensor measurments message data is written")
+      THEN("The write method reports ok status")
+      THEN("The body's header correctly stores the value of message type obtained from the message payload tag")
+      THEN("The output buffer is as expected")
+
+/* Write function for each of the message types (example: sensor_measurements)
+    WHEN("The payload is a sensor measurements message and write method is called")
+    // setup: sensor_measurements.flow = 50
+      THEN("The write method reports ok status")
+      THEN("The body's header correctly stores the value of message type obtained from the message payload tag")
+      THEN("The output buffer is as expected")
+/*
+
+/* Parse function
+
+Scenario: Message: The parse method correctly updates internal message type and sets the message payload fields after decoding from the input buffer
+
+  GIVEN("A Message object constructed with StateSegment Taggedunion, default message types and a max size of 252 bytes")
+
+    WHEN("A body with an encoded payload and 1 byte header of value equal to 0 is parsed")
+      THEN("The parse method reports invalid type status")
+      THEN("After the parse method is called, The value assigned to the type member is equal to the input_buffer body's header")
```

Comment:
The header will be changed to add a reserved field for future use, so this THEN should specify that it's comparing against the type field of the header.